### DISCUSSION TITLE
Is there a open continuation of this for gpt 3.5/4/ llama3.1/o1 ?

### DISCUSSION THREAD

**Discussion by manuvector** [karpathy/llm.c#790] [General]
Distance between open WEIGHT and frontier models looks to be stable, but open source? Any available training of gpt3.5 here or somewhere? 
I see only until gpt2 here, we're getting left behind 
Thanks

**Comment by linux-leo** [comment]
there is code for training much smaller versions of gpt3, but reproducing gpt4, o1 or the full llama model was never the point, this repository is meant to be a practical reference implementation to learn from and train with on a budget. But the price to reproduce any those models would be astronomically high, so implementing these architectures is expanding the scope of the project and therefore a slow process.

**Comment by gruai** [comment]
[Koifish](https://github.com/gruai/koifish)  needs only one day to train GPT2-1558M model on single 4090. 
It's next goal is llama3/DeepSeek.  Based on my development practice on Koifish, it's possible to train larger(2B~4B) models on single GPU card.
